src/Features/Core/Portable/FindUsages/AbstractFindUsagesService.ProgressAdapter.cs

```diff
@@ -107,20 +108,26 @@ public async ValueTask OnDefinitionFoundAsync(SymbolGroup group, CancellationTok
         }
 
         public async ValueTask OnReferencesFoundAsync(
-            ImmutableArray<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references, CancellationToken cancellationToken)
+            IAsyncEnumerable<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references, CancellationToken cancellationToken)
         {
-            foreach (var (group, _, location) in references)
-            {
-                var definitionItem = await GetDefinitionItemAsync(group, cancellationToken).ConfigureAwait(false);
-                var referenceItem = await location.TryCreateSourceReferenceItemAsync(
-                    classificationOptions,
-                    definitionItem,
-                    includeHiddenLocations: false,
-                    cancellationToken).ConfigureAwait(false);
-
-                if (referenceItem != null)
-                    await context.OnReferenceFoundAsync(referenceItem, cancellationToken).ConfigureAwait(false);
-            }
+            await ProducerConsumer<SourceReferenceItem>.RunParallelAsync(
```

Comment:
this logic is the major win.  allowing us to be computing the info for hte fuond references in parallel, sending out chunks of them to the client as we can.  while a chunk is in flight, we continually compute and buffer more.  sending the next chunk once we can.